### DISCUSSION TITLE
Got FutureWarning about _plotting.py

### DISCUSSION THREAD

**Discussion by kaya2k** [kernc/backtesting.py#705] [Q&A] [answered]
Hi, I got hundreds of this same warning using `backtest.plot` method.

`.../venv/lib/python3.8/site-packages/backtesting/_plotting.py:148:`
`FutureWarning: Passing method to Int64Index.get_loc is deprecated and will raise in a future version. Use index.get_indexer([item], method=...) instead. new_bar_idx = new_index.get_loc(mean_time, method='nearest')`

How can I remove this warning? Is there a way to remove it without modifying `.../backtesting/_plotting.py` file?

I'm currently using Python 3.8.9.

Thanks.

**Comment by kernc** [comment, marked as answer]
> Is there a way to remove it without modifying `.../backtesting/_plotting.py` file?

Eventually, we'll have to remove it from the file to avoid using the deprecated code. But you can silence the warning by placing:

```py
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
```
somewhere above your `backtest.plot()` call, or by setting the [environment variable `PYTHONWARNINGS`](https://docs.python.org/3/library/warnings.html#describing-warning-filters):
```bash
PYTHONWARNINGS=ignore::FutureWarning
```
before starting the interpreter.

**Reply by kaya2k** [reply]
I was using `warnings.simplefilter(action='ignore')`, but didn't know that I could give `category=FutureWarning` option.

Thank you, sir.
